File: src/main/java/ru/citeck/idea/templates/project/CiteckMavenArchetype.java
package ru.citeck.idea.templates.project;

import com.intellij.openapi.extensions.ExtensionPointName;
import com.intellij.util.xmlb.annotations.Attribute;
import org.jetbrains.idea.maven.model.MavenArchetype;

public class CiteckMavenArchetype {

    public static final ExtensionPointName<CiteckMavenArchetype> EP_NAME =
        ExtensionPointName.create("ru.citeck.idea.citeckMavenArchetype");

    @Attribute("displayName")
    public String displayName;

    @Attribute("groupId")
    public String groupId;

    @Attribute("artifactId")
    public String artifactId;

    @Attribute("version")
    public String version;

    @Attribute("description")
    public String description;

    @Attribute("isMicroservice")
    public Boolean isMicroservice;

    public MavenArchetype getMavenArchetype() {
        return new MavenArchetype(groupId, artifactId, version, null, description);
    }

    @Override
    public String toString() {
        return displayName;
    }

}

File: src/main/kotlin/ru/citeck/idea/newproject/CiteckMavenArchetype.kt
package ru.citeck.idea.newproject

import com.intellij.openapi.extensions.ExtensionPointName
import com.intellij.util.xmlb.annotations.Attribute
import org.jetbrains.idea.maven.model.MavenArchetype

class CiteckMavenArchetype {

    companion object {
        val EP_NAME: ExtensionPointName<CiteckMavenArchetype> =
            ExtensionPointName.create("ru.citeck.idea.citeckMavenArchetype")
    }

    @Attribute("displayName")
    lateinit var displayName: String

    @Attribute("groupId")
    lateinit var groupId: String

    @Attribute("artifactId")
    lateinit var artifactId: String

    @Attribute("version")
    lateinit var version: String

    @Attribute("description")
    var description: String? = null

    @Attribute("isMicroservice")
    var isMicroservice: Boolean = false

    fun getMavenArchetype(): MavenArchetype {
        return MavenArchetype(groupId, artifactId, version, null, description)
    }

    override fun toString(): String {
        return displayName
    }
}
